service/storage_service.cc

```diff
@@ -5202,6 +5209,18 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
             co_return;
         }
 
+        if (locator::contains(tinfo.replicas, dst.host)) {
+            throw std::runtime_error(format("Tablet {} has replica on {}", gid, dst.host));
+        }
+        auto src_dc_rack = get_token_metadata().get_topology().get_location(src.host);
+        auto dst_dc_rack = get_token_metadata().get_topology().get_location(dst.host);
+        if (src_dc_rack.dc != dst_dc_rack.dc) {
+            throw std::runtime_error(format("Attempted to move tablet {} between DCs ({} and {})", gid, src_dc_rack.dc, dst_dc_rack.dc));
+        }
+        if (src_dc_rack.rack != dst_dc_rack.rack && increase_replicas_per_rack(get_token_metadata().get_topology(), tinfo, dst_dc_rack.rack)) {
+            throw std::runtime_error(format("Attempted to move tablet {} between racks ({} and {})", gid, src_dc_rack.rack, dst_dc_rack.rack));
```

Comment:
Need to adjust the message so that it doesn't suggest that moving between racks is not allowed but rather HA would be reduced.

Reply:
This comment doesn't look like it was addressed.

Reply:
Right, sorry, I misunderstood